docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,374 @@
+
+sudo docker pull "$FROM_REPOSITORY/$IMAGE"
+sudo docker tag "$FROM_REPOSITORY/$IMAGE" "$TO_REPOSITORY/$IMAGE"
+sudo docker push "$TO_REPOSITORY/$IMAGE"
+```
+
+Repeat the pull, tag and push steps for all required images.
+
+### Images Option C: Side-load images
+
+Image side-loading is the process of loading all required OCI images directly
+into the container runtime, so that they do not have to be fetched at runtime.
+Upon choosing this option, you need to create a bundle of all the OCI images
+that will be used by the cluster.
+
+#### Create a bundle
+
+From a running cluster you may import images with `ctr`.
+Ensure you grab all images from all nodes in the cluster.
+  
+If we have an OCI image called nginx.tar,
+we can load this to all the cluster nodes by running the following command
+on any of them:
+
+```bash
+sudo k8s ctr image import - < nginx.tar
+```
+<!-- TODO: I think ctr is already installed where? -->
+On success, the output will look like this:
+
+```bash
+unpacking docker.io/library/nginx:latest \
+(sha256:9c58d14962869bf1167bdef6a6a3922f607aa823196c392a1785f45cdc8c3451)...d
+```
+
+For all standard OCI images that you will use, from .tar archives.
+
+## Deploy Canonical Kubernetes
+
+Now that you have fulfilled all steps in preparation for your
+air gapped cluster, it is time to get it deployed.
+
+### Step 1: Install Canonical Kubernetes
+
+Copy the `k8s.snap`, `k8s.assert`, `core20.snap` and `core20.assert` files into
+the target node, then install with:
+
+```bash
+sudo snap ack core20.assert && sudo snap install ./core20.snap
+sudo snap ack k8s.assert && sudo snap install ./k8s.snap --classic
+```
+
+Repeat the above for all nodes of the cluster.
+
+### Step 2: Form Canonical Kubernetes cluster
+
+Now are ready to bootstrap the cluster by running:
+
+```bash
+sudo k8s bootstrap
+```
+
+```{note}  Please skip the following section for one node deployments. ```
+
+You can add and remove nodes as described in the
+[add-and-remove-nodes tutorial][nodes].
+
+After a while, confirm that all the cluster nodes show up in
+the output of the `sudo k8s kubectl get node`. 
+
+The nodes will most likely be in `NotReady` state,
+since we still need to ensure the container runtime can fetch images.
+
+### Step 3: Container Runtime
+
+#### Container Runtime Option A: Configure HTTP proxy for registries
+
+Edit `/etc/environment` and set the appropriate http_proxy, https_proxy and
+no_proxy variables as described in the
+[adding proxy configuration section][proxy].
+
+Then restart the k8s snap with:
+
+```bash
+sudo snap restart k8s
+```
+
+#### Container Runtime Option B: Configure registry mirrors
+
+This requires that you have already setup a registry mirror,
+as explained in the preparation section on the private registry mirror.
+
+Assuming the registry mirror is at 10.100.100.100:5000, edit 
+`/var/snap/k8s/common/etc/containerd/hosts.d/docker.io/hosts.toml`
+and make sure it looks like this:
```

Comment:
Explain that for each upstream registry that a mirror will be used, we need to create a `hosts.toml` file. Then, describe that this example documents configuring `http://10.100.100.100:5000` as a mirror for `docker.io`